Worklet Improvents / fixes

daslyfe opened the pull request:
These are fixes for the shape/crush/coarse worklets

- all are now stereo fx
- shape now has an optional "postgain" param when using array syntax: "shape("8:.5")"
- coarse works predictably on non chromium
- coarse and crush fx no longer go silent when inputting 0
- Added a distort worklet, which is the same as shape but with a much safer/gradual input curve

daslyfe commented:
> nice! is the shape range now very different / would this change existing patterns using shape audibly?
yes before 0 -.8 was the usable range.  Now it is 0-10 and the exponential curve of the shape param input is different/more predictable.

daslyfe commented:
> yep shape is dangerous :D I just worry about the breaking change, so maybe find a new name and keep the existing shape as is? iirc the shape of tidal also has a range between 0 and 1 with a similar explosive behavior. not sure if i like the range between 0 and 10 because it does not play as well with normalized values.

I adjusted it so 0 - 1 goes from not distorted to very distorted in a linear fashion. if you want to be more extreme you can continue to go above 1. Would that be a decent compromise? It could still affect the sound of patterns that use shape slightly but it makes the function a lot more useful imo. I can also create another version called shape2 or something like that :)

felixroos commented:
I also agree that a more linear distortion curve is very desirable, it might still make sense to add it as an alias because it changes the mix of a lot of existing stuff + keeping parity with tidal is generally a good thing. Maybe just call it `distort` or `shapelin` (or both) ?

daslyfe commented:
> I also agree that a more linear distortion curve is very desirable, it might still make sense to add it as an alias because it changes the mix of a lot of existing stuff + keeping parity with tidal is generally a good thing. Maybe just call it `distort` or `shapelin` (or both) ?

Okay I updated it and added another processor to keep shape behavior the same. New processor is called distort, but has a shorthand of dist

daslyfe commented on packages/superdough/superdough.mjs:
```diff
@@ -457,7 +458,14 @@ export const superdough = async (value, deadline, hapDuration) => {
   crush !== undefined && chain.push(getWorklet(ac, 'crush-processor', { crush }));
-  shape !== undefined && chain.push(getWorklet(ac, 'shape-processor', { shape }));
+  if (shape !== undefined) {
+    const input = Array.isArray(shape) ? { shape: shape[0], postgain: shape[1] } : { shape };
```
cool, I updated it but instead of using "postgain" I used shapevol and distortvol because postgain is already the name of a different control.

daslyfe replied:
this is ready to go again
